cli/src/test/java/com/devonfw/tools/ide/commandlet/InstallCommandletTest.java

```diff
@@ -0,0 +1,115 @@
+
+    Path windowsFilePath = resourcePath.resolve("__files").resolve("java-17.0.6-windows-x64.zip");
+    String windowsLength = String.valueOf(Files.size(windowsFilePath));
+    server.stubFor(
+        get(urlPathEqualTo("/installTest/windows")).willReturn(aResponse().withHeader("Content-Type", "application/zip")
+            .withHeader("Content-Length", windowsLength).withStatus(200).withBodyFile("java-17.0.6-windows-x64.zip")));
+
+    Path linuxFilePath = resourcePath.resolve("__files").resolve("java-17.0.6-linux-x64.tgz");
+    String linuxLength = String.valueOf(Files.size(linuxFilePath));
+    server.stubFor(
+        get(urlPathEqualTo("/installTest/linux")).willReturn(aResponse().withHeader("Content-Type", "application/tgz")
+            .withHeader("Content-Length", linuxLength).withStatus(200).withBodyFile("java-17.0.6-linux-x64.tgz")));
+
+    server.stubFor(
+        get(urlPathEqualTo("/installTest/macOS")).willReturn(aResponse().withHeader("Content-Type", "application/tgz")
+            .withHeader("Content-Length", linuxLength).withStatus(200).withBodyFile("java-17.0.6-linux-x64.tgz")));
+
+  }
+
+  /**
+   * Test of {@link InstallCommandlet} run, when Installed Version is null.
+   */
+  @Test
+  public void testInstallCommandletRunWithVersion() throws IOException {
+
+    // arrange
+    String path = "workspaces/foo-test/my-git-repo";
+    IdeContext context = newContext("basic", path, true);
+    InstallCommandlet install = context.getCommandletManager().getCommandlet(InstallCommandlet.class);
+    install.tool.setValueAsString("java");
+    mockWebServer();
+    // act
+    install.run();
+    // assert
+    assertTestInstall(context);
+
+  }
+
+  /**
+   * Test of {@link InstallCommandlet} run, when Installed Version is set.
+   */
+  @Test
+  public void testInstallCommandletRunWithVersionAndVersionIdentifier() throws IOException {
+
+    // arrange
+    String path = "workspaces/foo-test/my-git-repo";
+    IdeContext context = newContext("basic", path, true);
+    InstallCommandlet install = context.getCommandletManager().getCommandlet(InstallCommandlet.class);
+    install.tool.setValueAsString("java");
+    install.version.setValueAsString("17.0.6");
+    mockWebServer();
+
+    // act
+    install.run();
+    // assert
+    assertTestInstall(context);
+
+  }
+
+  public void assertTestInstall(IdeContext context) {
+
+    assertThat(context.getSoftwarePath().resolve("java")).exists();
+    assertThat(context.getSoftwarePath().resolve("java/InstallTest.txt")).hasContent("This is a test file.");
+    assertThat(context.getSoftwarePath().resolve("java/bin/HelloWorld.txt")).hasContent("Hello World!");
+    if(context.getSystemInfo().isWindows()){
+      assertThat(context.getSoftwarePath().resolve("java/Java.cmd")).exists();
+    } else if (context.getSystemInfo().isLinux() || context.getSystemInfo().isMac()) {
+      assertThat(context.getSoftwarePath().resolve("java/Java.sh")).exists();
+    }
```

Comment:
My suggestion was to include commands following the actual tool conventions so we can not only test the installation but also the execution (there is some non-trivial logic to find the tool executable to run after the installation). Then we could actually run `Java` commandlet and if such `java/bin/java.cmd` MS-Dos script of `java/bin/java` as Bash-Script is present then the proper dummy output could be asserted depending on the OS (e.g. `Dummy Java 17.0.6 on Windows` could be asserted on Windows if you put `echo Dummy Java 17.0.6 on Windows` into `java.cmd`).

Reply:
After further analysis, the script using echo is not writing into the system.out, and running the test manually I receive the following warning and report: 

`[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file C:\Users\molanger\ide\workspaces\main\127\IDEasy\cli\target\surefire-reports\2023-12-14T12-15-45_934-jvmRun1.dumpstream`

` Created at 2023-12-14T12:15:47.165
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'C:\Users\molanger\ide\workspaces\main\127\IDEasy\cli>echo "This is a test install file." '.

# Created at 2023-12-14T12:15:47.174
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '"This is a test install file."'.

`

Reply:
I am unclear what is going on but my first guess is that maven surefire is somehow interfering here and already redirecting streams of our JUnit test that again wanted to do the same.
So lets first get this merged and then consider further perfection...